Src/AmrCore/AMReX_FillPatcher.H

```diff
@@ -330,14 +385,202 @@ FillPatcher<MF>::fillCoarseFineBoundary (MF& mf, IntVect const& nghost, Real tim
                         ncomp, IntVect(0), m_cgeom, m_fgeom,
                         amrex::grow(amrex::convert(m_fgeom.Domain(),
                                                    mf.ixType()),nghost),
-                        ratio, m_interp, bcs, bcscomp);
+                        m_ratio, m_interp, bcs, bcscomp);
 
         post_interp(*m_cf_fine_data, scomp, ncomp);
 
         mf.ParallelCopy(*m_cf_fine_data, scomp, dcomp, ncomp, IntVect{0}, nghost);
     }
 }
 
+template <typename MF>
+template <std::size_t order>
+void FillPatcher<MF>::storeRKCoarseData (Real /*time*/, Real dt, MF const& S_old,
+                                         Array<MF,order> const& RK_k)
+{
+    m_dt_coarse = dt;
+    m_cf_crse_data.resize(order+1);
+
+    auto const& fpc = getFPinfo();
+
+    for (auto& tmf : m_cf_crse_data) {
+        tmf.first = std::numeric_limits<Real>::lowest(); // because we dont' need it
+        tmf.second = std::make_unique<MF>(make_mf_crse_patch<MF>(fpc, m_ncomp));
+    }
+    m_cf_crse_data[0].second->ParallelCopy(S_old, m_cgeom.periodicity());
+    for (std::size_t i = 0; i < order; ++i) {
+        m_cf_crse_data[i+1].second->ParallelCopy(RK_k[i], m_cgeom.periodicity());
+    }
+}
+
+template <typename MF>
+template <typename BC>
+void FillPatcher<MF>::fillRK (int stage, int iteration, int ncycle,
+                              MF& mf, Real time, BC& cbc, BC& fbc,
+                              Vector<BCRec> const& bcs)
+{
+    int rk_order = m_cf_crse_data.size()-1;
+    if (rk_order != 3 && rk_order != 4) {
+        amrex::Abort("FillPatcher: unsupported RK order "+std::to_string(rk_order));
+        return;
+    }
+    AMREX_ASSERT(stage > 0 && stage <= rk_order);
```

Comment:
Is `storeRKCoarseData` templated on the order to make it easier on the compiler? It looks like `fillRK` can deduce the Runge Kutta order from FillPatcher member objects. Is this a safe assumption, and if so, can it be made in both functions for consistency?

Reply:
`storeRKCoarseData` is templated not for the benefit of the compiler.  It it done this way trying to make the user less likely to make a mistake, because it won't compile if the user calls `RungeKutta::RK3(..., [&] (Array<MultiFab,4> const& rkk) {...});`.  If it's not a template, then the user will be asked to provide a `Vector<MultiFab>`.  Then the assertion on the size can only be done at run time.  As for `fillRK`, it's called after `storeRKCoarseData` and it does not have any parameters that depend on the order.